magma/backend/coreir_transformer.py

```diff
@@ -21,17 +21,49 @@
 from ..passes import InstanceGraphPass
 from ..tuple import Tuple
 from .util import get_codegen_debug_info
+from magma.clock import ClockTypes, first, get_first_clock, wire_clock_port
 from magma.config import get_debug_mode
-
+from magma.protocol_type import MagmaProtocol, MagmaProtocolMeta
 from magma.ref import PortViewRef, ArrayRef
-from magma.clock import ClockTypes, first, get_first_clock, wire_clock_port
 
 
 # NOTE(rsetaluri): We do not need to set the level of this logger since it has
 # already been done in backend/coreir_.py.
 _logger = root_logger().getChild("coreir_backend")
 
 
+def _collect_drivers(value):
+    """
+    Iterate over value to collect the child drivers, packing slices together
+    """
+    drivers = []
+    start_idx = 0
+    for i in range(1, len(value)):
+        # If the next value item is not a reference to an array of bits where
+        # the array matches the previous item and the index is incremented by
+        # one, append the current slice to drivers (may introduce slices of
+        # length 1)
+        if not (
+            isinstance(value[i].name, ArrayRef) and
+            issubclass(value[i].name.array.T, Digital) and
+            isinstance(value[i - 1].name, ArrayRef) and
+            value[i].name.array is value[i - 1].name.array and
+            value[i].name.index == value[i - 1].name.index + 1
+        ):
+            drivers.append(value[start_idx:i])
+            start_idx = i
+    drivers.append(value[start_idx:])
+    return drivers
+
+
+def _unwrap(x):
```

Comment:
Maybe this should be in some sort of util file or in the protocol_type module it self.  There is an effectively equivalent function here
https://github.com/phanrahan/magma/blob/master/magma/syntax/sequential.py#L21

Reply:
when we do a nuclear "deprotocol" pass I think that will consume all of this

Reply:
Im not sure.  Wouldn't the use in sequential still be necessary?

Reply:
yeah that's fair. think it would be a good idea to separate out. already merged to accelerate unblocking, so I will come back and make a PR for this.

Reply:
Oh for sure I didn't think the PR should be blocked on this.  Just a thought